### DISCUSSION TITLE
LEFT and RIGHT functions are returning incorrect results

### DISCUSSION THREAD

**Discussion by dksingla** [handsontable/hyperformula#959] [API Questions]
### Description

Below is an example of the LEFT formula

[["0003", "=LEFT(A1,2)"]];

It returns 3, instead, it should return 00 as LEFT takes the first argument of the typed string.

**Comment by sequba** [comment]
Hi @dksingla thanks for your report. This behaviour is caused by interpreting string `0003` as a number and therefore trimming the leading zeros. To avoid this behaviour you need to start your string literal with `'` character (apostrophe) to inform the parser that you don't want it to be converted to number.

Try this:
```
[["'0003", "=LEFT(A1,2)"]];
```

I see that this behaviour is not documented in our docs, so I'm creating an issue to do so: #960

Unit tests that demonstrate this: 
https://github.com/handsontable/hyperformula/blob/fd0a40581d1dea2a19c2307228825ad0c3b6dedb/test/CellContentParser.spec.ts#L128-L135

**Reply by dksingla** [reply]
Thanks, @sequba for the explanation.

**Comment by sequba** [comment]
@dksingla We've just released the new version of the documentation with a [detailed description of how to force a string value type](https://hyperformula.handsontable.com/guide/types-of-values.html#forcing-the-string-value-type).
